core/src/com/mygdx/game/models/Command.java

```diff
@@ -1,14 +1,93 @@
 package com.mygdx.game.models;
 
-public interface Command {
+/**
+ * Class to represent a command
+ */
+public class Command {
+    /* Type of action the command represents */
+    private String actionType;
 
-    /**
-     * Get the action associated with the command
-     */
-    String getAction();
+    /* Acting unit's ID */
+    private int unitId;
 
-    /**
-     * Get position that the action is directed at
-     */
-    Position getPosition();
+    /* Target unit's ID */
+    private int targetUnitId;
+
+    /* Position action is taken from */
+    private Position startPosition;
```

Comment:
You know the start position because you know the unit that is preforming the action. therefore this might not be necessary.

Reply:
Removed